src/OrleansProviders/Streams/Common/PooledCache/FixedSizeBuffer.cs

```diff
@@ -77,8 +77,13 @@ public bool TryGetSegment(int size, out ArraySegment<byte> value)
         /// </summary>
         public override void SignalPurge()
         {
-            count = 0;
             purgeAction?.Invoke(this);
+        }
+
+        /// <inheritdoc />
+        public override void OnResetState()
```

Comment:
because now SignalPurge doesn't mean the buffer will be returning to the pool, so reset state in SignalPurge is not valid anymore. I need to clean it up in FixedSizeBuffer.Dispose method